### ISSUE TITLE
Helper for launching `redis-server`?

### ISSUE THREAD

**Issue by djpalumbo** [saiichihashimoto/redis-prebuilt#29]
The [`mongodb-prebuilt`](https://www.npmjs.com/package/mongodb-prebuilt) package has `MongodHelper` to allow for `mongod` to be run from within your Node app. This is an awesome feature and I was wondering if there's any plan to introduce something similar to your package in order to launch `redis-server`.

I'd really like to be able to run this in the background before I do other things, for example:
```js
// Side note: This require fails without 'main' declared in package.json
const { RedisServerHelper } = require('redis-prebuilt');

let redisServerHelper = new RedisServerHelper(['--port', "6379"]);
 
redisServerHelper.run().then((started) => {
    console.log('redis-server is running');
}, (e) => {
    console.log('error starting', e);
});
```

**Comment by saiichihashimoto** [comment]
Yes, this is something I was thinking of doing. Initially, I wanted a redis version of mongodb-prebuilt's cli API, but intend to have full parity. I'm not sure what my time schedule is, so PRs are welcome as long as they don't decrease test coverage and fit the linting rules.

**Comment by djpalumbo** [comment]
I've started a [fork of your repo](https://github.com/djpalumbo/redis-prebuilt). I haven't written any tests yet, but what would be the best way would be for me to monitor test coverage?

**Comment by saiichihashimoto** [comment]
I believe the test command should just do it? If not, `--coverage` should do it. The test command is just running jest.

**Comment by djpalumbo** [comment]
Thanks! I didn't realize Jest has options for coverage (only used Mocha before). I think I've got the hang of it.

**Comment by djpalumbo** [comment]
Hey there. I'm sure you're busy and I don't mean to rush you, but do you know when you'll be able to take a look at PR #33 ?

**Comment by saiichihashimoto** [comment]
Hey @djpalumbo! I'm currently boarding a flight to Japan for a week and a half so... it'll be a while :-(

**Comment by djpalumbo** [comment]
Oof :/ I understand. Just for some context‒ I'm trying to include your package (with this enhancement) in my final undergrad CS project that I'll be presenting in a little over two weeks. It would really make a world of difference if I could add it in sooner rather than later. If not, I can probably use `execa` instead, but I sure would like to include `redis-prebuilt`.

**Comment by saiichihashimoto** [comment]
Yeah, I'm not sure what to say. For some reason, Travis didn't run the build for your fork so the automated tests aren't running. I didn't bring my laptop so I can't really do any of this work, I'm using wifi on a plane with my phone right now. Sorry!

**Comment by saiichihashimoto** [comment]
You could use your fork directly as the dependency in the time being.

**Comment by djpalumbo** [comment]
Do you mean by adding it as a package.json dependency like so? :
```js
"dependencies": {
    "redis-prebuilt": "git+https://git@github.com/djpalumbo/redis-prebuilt.git",
},
```
Or would you suggest publishing directly to NPM?

**Comment by saiichihashimoto** [comment]
The former. You may need to add a postinstall step to build it, since the Travis build usually builds it and publishes the built files. Mess with it a bit. Sorry for the inconvenience!

**Comment by djpalumbo** [comment]
Looks like I'm all set for the time being. All I needed to do was [add the `"prepare"` hook](https://github.com/djpalumbo/redis-prebuilt/commit/3df45853b81f647b215e68f53f97dc300ad43000) to the package.json scripts. Still, I hope the feature is good enough that it can be merged and published before I wrap up work on my project. I'll bump this thread in a week or two to drop you a friendly reminder. Safe travels and I hope you enjoy your time in Japan!

**Comment by saiichihashimoto** [comment]
Hey @djpalumbo! I dropped some comments on #33, if you wanted to take a look. If you're done working on this let me know and I can get to it on my own time.

**Comment by djpalumbo** [comment]
Thanks for the review @saiichihashimoto! I've been caught up with final exams, but I should be able to take a look some time next week. Of course, you're more than welcome to make changes if you're ready.
